src/utils/fetchData.ts

```diff
@@ -0,0 +1,37 @@
+import { Platform } from 'react-native';
+
+export function fetchText(uri?: string): Promise<string | null> {
+  if (
+    uri &&
+    uri.startsWith('data:image/svg+xml;utf8') &&
+    Platform.OS === 'android'
+  ) {
+    return new Promise((resolve) => {
+      resolve(dataUriToXml(uri));
+    });
+  } else {
+    return fetchUriData(uri);
+  }
+}
+
+function dataUriToXml(uri: string): string | null {
+  try {
+    // decode and remove data:image/svg+xml;utf8, prefix
+    const xml = decodeURIComponent(uri).split(',').slice(1).join(',');
+    return xml;
+  } catch (error) {
+    console.log('error', error);
```

Comment:
We probably should throw an `Error` here instead of `log`. Also, it should be more descriptive like here https://github.com/software-mansion/react-native-svg/pull/2396/files#diff-1d505e3c2299d034f80dac76837cf7c866616fd1a8c2361295d141833b104f46R36

Reply:
done.